iis-workflows/iis-workflows-ingest-pmc/src/main/java/eu/dnetlib/iis/workflows/ingest/pmc/metadata/MetadataImporter.java

```diff
@@ -46,15 +85,29 @@ protected void map(AvroKey<DocumentText> key, NullWritable value, Context contex
 				saxParser.parse(new InputSource(
 						new StringReader(nlm.getText().toString())), 
 						pmcXmlHandler);
-                context.write(new AvroKey<ExtractedDocumentMetadata>(output.build()), 
-                		NullWritable.get());
+				mos.write(namedOutputMeta, 
+						new AvroKey<ExtractedDocumentMetadata>(output.build()));
         	 } catch (ParserConfigurationException e) {
-             	throw new IOException("Text extraction failed for id " + nlm.getId() + 
-             			" and text: " + nlm.getText(), e);
+        		 Map<CharSequence, CharSequence> auditSupplementaryData = new HashMap<CharSequence, CharSequence>();
+        			auditSupplementaryData.put(FAULT_TEXT, nlm.getText());
```

Comment:
Ok, It's not that easy as I thought. According to specification of [Avro schema resolution][1], in order to make a generic code read `MetadataExtractionFault` and `PMCIngestionFault` records as if each of them was a `Fault` record (polymorphism-like behaviour), both of these former records would have to be named `Fault` and come from the same namespace as `Fault`. 

I think that our code that generates java classes from Avro schemas wouldn't like this setting. Apart from that, it might be confusing to have different structures with the same name.

BTW, field ordering doesn't matter - according to [Avro schema resolution][1] "the ordering of fields may be different: fields are matched by name."

[1]: http://avro.apache.org/docs/1.7.7/spec.html#Schema+Resolution